crates/noirc_evaluator/src/ssa/node.rs

```diff
@@ -543,6 +547,7 @@ pub enum Operation {
     Call(noirc_frontend::node_interner::FuncId, Vec<NodeId>, Vec<(ArrayId, u32)>), //Call a function
     Return(Vec<NodeId>), //Return value(s) from a function block
     Result { call_instruction: NodeId, index: u32 }, //Get result index n from a function call
+    Cond { condition: NodeId, lhs: NodeId, rhs: NodeId },
```

Comment:
lhs and rhs aren't good names for these, lets rename them.

The name `Cond` is also ambiguous to me, I'm not sure if its meant as an `If` substitute, or if its short for conditional jump, or something else. Why is Cond needed in the IR at all? If expressions should already be encodeable via Jne/Jeq/Jmps and a Phi to merge the result.

Reply:
lhs and rhs are good names for this instruction which takes these two arguments and returns one of them depending on the condition.
`If` is indeed encoded with jumps and phis in a first step. Then we remove them thanks to this new instruction. 
The new instruction gives a better and simpler representation compared to extending the handling of new instructions which would require special handling in many places. This should be avoided when possible. For instance it won't be possible for function calls and they will need to be extended.

Reply:
lhs and rhs aren't good names because neither argument is inherently more left or right compared to the other. A better name would relate to which is chosen from which value of the condition. E.g. `then`/`otherwise`, `iftrue`/`iffalse`, etc.
Can you enlighten me on why function calls would need special handling? I can see us failing to inline them more often if users try using recursion with ifs that involve a witness, but this is expected to some degree (also something we should keep in mind for designing good error messages for in the future).

Reply:
We will inline after the IF reduction because function calls can be optimised if they are in both branches. But because they can have side effect, we must keep their assumption which will be used during inlining.

Reply:
lhs and rhs are fine because they left and right of the colon in the ternary operator. 
May be I can rename the Cond operation to Ternary if this is clearer for you.

Reply:
Noir has no ternary operator, I'd still prefer for their name to match their purpose rather than their position

Reply:
Noir could have a ternary operator and it now has a ternary ssa instruction.
Anyway, since you prefer another name, I did rename them

Reply:
Thanks for the rename - as for a ternary operator noir shouldn't need one since it's `if` can be used in an expression